docs/dev_guide/compare-prog-models.rst

```diff
@@ -0,0 +1,767 @@
+prior to ``__syncthreads()`` are visible to all threads in the block. The CUDA
+``__syncthreads`` function can be mapped to ``sycl::group_barrier`` (with a
+``sycl::group`` object passed in). For the CUDA ``__syncthreads_and``,
+``__syncthreads_or``, and ``__syncthreads_count`` function migration, an additional
+group algorithm is needed after ``sycl::group_barrier``. The CUDA  ``__syncwarp``
+function can be mapped to ``sycl::group_barrier``  (with a ``sycl::sub_group``
+object spassed in).
+
+The following table shows CUDA to SYCL mapping for synchronization functions:
+
+.. list-table::
+   :widths: 50 50
+   :header-rows: 1
+
+   * - CUDA
+     - SYCL
+   * - ``__syncthreads()``
+     - ``sycl::group_barrier(Group)``
+   * - ``__syncthreads_and()``
+     - ``sycl::group_barrier(Group)`` and ``sycl::all_of_group(Group, predicate)``
+   * - ``__syncthreads_or()``
+     - ``sycl::group_barrier(Group)`` and ``sycl::any_of_group(Group, predicate)``
+   * - ``__syncthreads_count()``
+     - ``sycl::group_barrier(Group)`` and ``sycl::reduce_over_group(Group, predicate?1:0, sycl::ext::oneapi::plus<>())``
+   * - ``__syncwarp()``
+     - ``sycl::group_barrier(Sub_group)``
+
+Memory Fence API
+****************
+
+Memory fence functions can be used to enforce some ordering on memory accesses.
+The memory fence functions differ in the scope in which the orderings are enforced.
+CUDA memory fence functions can be mapped to ``sycl::atomic_fence`` with different
+memory scope.
+
+The following table shows CUDA to SYCL mapping for fence functions:
+
+.. list-table::
+   :widths: 50 50
+   :header-rows: 1
+
+   * - CUDA
+     - SYCL
+   * - ``__threadfence_block()``
+     - ``sycl::atomic_fence(sycl::memory_order::acq_rel, sycl::memory_scope::work_group)``
+   * - ``__threadfence()``
+     - ``sycl::atomic_fence(sycl::memory_order::acq_rel, sycl::memory_scope::device)``
+   * - ``__threadfence_system()``
+     - ``sycl::atomic_fence(sycl::memory_order::acq_rel, sycl::memory_scope::system)``
+
+Warp Intrinsic API
+******************
+
+CUDA warp intrinsic functions, including warp vote and shuffle functions, can
+be mapped to the SYCL group algorithm API.
+
+Warp Vote API
+*************
+
+The CUDA warp vote API can map to the SYCL group algorithm API, as shown in the
+following table:
+
+.. list-table::
+   :widths: 50 50
+   :header-rows: 1
+
+   * - CUDA
+     - SYCL
+   * - ``__all()/__all_sync()``
+     - ``sycl::all_of_group()``
+   * - ``__any()/__any_sync()``
+     - ``sycl::any_of_group()``
+   * - ``__ballot()/__ballot_sync()``
+     - ``sycl::reduce_over_group()``
+
+For the sync version of CUDA warp intrinsic functions, a mask is passed that
+specifies the threads participating in the call. The equivalent SYCL API does not
+support mask directly. Refer to the reference implementation of mask version APIs
+in the `SYCLomatic project <https://github.com/oneapi-src/SYCLomatic>`_.

```

Comment:
the link should be https://github.com/oneapi-src/SYCLomatic/blob/4893685d6f2f1c6126a1962be65dfa7688bb4162/clang/runtime/dpct-rt/include/util.hpp.inc#L413

Reply:
Done.